ui/pages/notifications/notifications-list-read-all-button.tsx

```diff
@@ -28,38 +29,32 @@ export const NotificationsListReadAllButton = ({
   const t = useI18nContext();
   const { markNotificationAsRead } = useMarkNotificationAsRead();
   const trackEvent = useContext(MetaMetricsContext);
-
   const unreadNotifications = useSelector(getUnreadNotifications);
 
-  const [notificationReadArray, setNotificationReadArray] =
-    useState<MarkAsReadNotificationsParam>([]);
-
-  useEffect(() => {
+  const handleOnClick = () => {
     let notificationsRead: MarkAsReadNotificationsParam = [];
 
     if (notifications && notifications.length > 0) {
       notificationsRead = notifications
         .filter(
           (notification): notification is Notification =>
-            (notification as Notification).id !== undefined,
+            (notification as Notification).id !== undefined &&
+            notification.type !== SNAP,
         )
         .map((notification: Notification) => ({
           id: notification.id,
           type: notification.type,
           isRead: notification.isRead,
         }));
     }
-    setNotificationReadArray(notificationsRead);
-  }, [notifications]);
 
-  const handleOnClick = () => {
     trackEvent({
       category: MetaMetricsEventCategory.NotificationInteraction,
       event: MetaMetricsEventName.MarkAllNotificationsRead,
     });
 
     // Mark all metamask notifications as read
-    markNotificationAsRead(notificationReadArray);
+    markNotificationAsRead(notificationsRead);
```

Comment:
I got really confused since we were filtering out snap notifications.

But yes this method (as the comment above mentions) is only for wallet notifications, so we can filter out these notifications.